### PR TITLE
Update to Go 1.15.7

### PR DESCRIPTION
**Pull request by axw**
## Motivation/summary

Updates to latest beats and Go 1.15.7.

`*.ssl.verification_mode` docs have been updated in `apm-server.yml` to describe the possible modes. The TLS changes in Go 1.15 are relevant to verifying server (Elasticsearch, Kibana, etc.) certificates, so no need to update `apm-server.ssl.*`.

## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)
- [x] Documentation has been updated

## How to test these changes

Run the unit, system, integration tests.

## Related issues

Closes https://github.com/elastic/apm-server/issues/4649

### CODE REVIEW

**Comment by axw** [comment]
@simitt there's a couple of things you mentioned in #4649 that I have not done.

> Update in the apm-server.yml for apm-server.ssl. section (acting as server)
>
> Adapt tests where necessary:
>  - create certs with hostname in Subject Alternative Name fields and test behavior for strict mode
>  - keep some test for full mode

I don't believe these are necessary, as the certificate changes affect verification of server certificates. AFAICS we don't have any tests for outputs connecting over TLS. Nor do I think we should, since that is libbeat's domain.

**Review by jalvz — APPROVED** [review]

File: changelogs/head.asciidoc [hunk 1]
```diff
@@ -18,6 +18,7 @@ https://github.com/elastic/apm-server/compare/7.11\...master[View commits]
 * Jaeger gRPC is now served over the same port as the Elastic APM agent protocol {pull}4618[4618]
 * Support for reloading config in Fleet mode, gracefully stopping the HTTP server and starting a new one {pull}4623[4623]
 * Add a `_doc_count` field to transaction histogram docs {pull}4647[4647]
+* Upgrade Go to 1.14.12 {pull}TODO[TODO]
```
Comment (simitt) [inline changelogs/head.asciidoc:21]: This should be `1.15.7`.

Comment (axw) [reply changelogs/head.asciidoc:21]: :facepalm: https://github.com/elastic/apm-server/pull/4667

**Comment by kuisathaverat** [comment]
this causes the following error on the apm-integration-testing

```
[2021-02-02T14:33:32.856Z] ./build/linux/bin/mage -compile=build/linux/mage
[2021-02-02T14:33:54.789Z] # github.com/elastic/beats/v7/libbeat/common/transport/tlscommon
[2021-02-02T14:33:54.790Z] /go/pkg/mod/github.com/elastic/beats/v7@v7.0.0-alpha2.0.20210127154636-fe8f75895aa1/libbeat/common/transport/tlscommon/tls_config.go:112:3: unknown field 'VerifyConnection' in struct literal of type tls.Config
[2021-02-02T14:33:54.790Z] /go/pkg/mod/github.com/elastic/beats/v7@v7.0.0-alpha2.0.20210127154636-fe8f75895aa1/libbeat/common/transport/tlscommon/tls_config.go:123:4: unknown field 'VerifyConnection' in struct literal of type tls.Config
[2021-02-02T14:33:54.790Z] /go/pkg/mod/github.com/elastic/beats/v7@v7.0.0-alpha2.0.20210127154636-fe8f75895aa1/libbeat/common/transport/tlscommon/tls_config.go:141:4: unknown field 'VerifyConnection' in struct literal of type tls.Config
[2021-02-02T14:33:54.790Z] /go/pkg/mod/github.com/elastic/beats/v7@v7.0.0-alpha2.0.20210127154636-fe8f75895aa1/libbeat/common/transport/tlscommon/tls_config.go:149:8: config.VerifyConnection undefined (type *tls.Config has no field or method VerifyConnection)
[2021-02-02T14:33:54.790Z] note: module requires Go 1.15
[2021-02-02T14:34:07.043Z] Error: error compiling magefiles
[2021-02-02T14:34:07.043Z] make: Leaving directory '/go/src/github.com/elastic/apm-server'
[2021-02-02T14:34:07.043Z] make: *** [Makefile:256: build/linux/mage] Error 1
```

**Comment by axw** [comment]
Sorry @kuisathaverat, I forgot about updating apm-integration-testing. Thanks for fixing.

**Pull request merged**